CMakeLists.txt

```diff
@@ -38,6 +38,10 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
 	else()
 		message(FATAL_ERROR "Your C++ compiler is too old, please upgrade.")
 	endif()
+else()
+  set(CMAKE_CXX_STANDARD 17)
```

Comment:
This is not really the way to do this (neither is the way we're using now though but I'm looking to fix it in a future PR)

Reply:
it's definitely better than setting compiler flags manually, which is currently done for gcc and clang

Reply:
Which is what I said. I've already made this somewhat better in s private branch but haven't committed it because it's sorta wrapped in a larger issue.